server/src/test/java/io/crate/planner/DeletePlannerTest.java

```diff
@@ -79,6 +80,15 @@ public void testBulkDeletePartitionedTable() throws Exception {
         assertThat(partitionSymbols).satisfiesExactly(exactlyInstanceOf(ParameterSymbol.class));
     }
 
+    // bug: https://github.com/crate/crate/issues/14347
+    @Test
+    public void test_delete_plan_with_where_clause_involving_pk_and_non_pk() throws Exception {
+        Plan delete = e.plan("delete from users where id in (1,2,3) and name='dummy'");
+        assertThat(delete instanceof DeletePlanner.Delete).isTrue();
```

Comment:
```suggestion
        assertThat(delete).isInstanceOf(DeletePlanner.Delete.class);
```

Reply:
thx! `.isExactlyInstanceOf` to be even safer and consistent,